File: android/app/src/main/java/com/rork/porchivo/ui/screens/PorchPartnerScreen.kt
package com.rork.porchivo.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.outlined.LocationOn
import androidx.compose.material.icons.outlined.BackHand
import androidx.compose.material.icons.outlined.Paid
import androidx.compose.material.icons.outlined.VerifiedUser
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.NavController
import com.rork.porchivo.model.ShipmentStatus
import com.rork.porchivo.ui.components.EmptyState
import com.rork.porchivo.ui.components.ShipmentCard
import com.rork.porchivo.ui.navigation.Routes
import com.rork.porchivo.ui.theme.PorchivoTheme
import com.rork.porchivo.ui.viewmodel.AppViewModel
import com.rork.porchivo.ui.viewmodel.ShipmentsViewModel

@Composable
fun PorchPartnerScreen(
    navController: NavController,
    modifier: Modifier = Modifier,
    appViewModel: AppViewModel = viewModel(),
    shipmentsViewModel: ShipmentsViewModel = viewModel(),
) {
    val c = PorchivoTheme.colors
    val myShipments by shipmentsViewModel.myShipments.collectAsStateWithLifecycle()

    val activeHolds = myShipments.filter { it.status == ShipmentStatus.ACCEPTED }

    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .background(c.background),
        contentPadding = androidx.compose.foundation.layout.PaddingValues(
            start = 16.dp, end = 16.dp, top = 16.dp, bottom = 24.dp,
        ),
        verticalArrangement = Arrangement.spacedBy(14.dp),
    ) {
        item {
            Text(
                text = "Porch Partner",
                color = c.textPrimary,
                fontSize = 26.sp,
                fontWeight = FontWeight.Black,
            )
        }
        item {
            Text(
                text = "Earn by helping neighbors",
                color = c.textSecondary,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
            )
        }
        item {
            EarningsCard()
        }
        item {
            Text(
                text = "Active Holds",
                color = c.textPrimary,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
            )
        }
        if (activeHolds.isEmpty()) {
            item {
                EmptyState(
                    icon = Icons.Outlined.BackHand,
                    title = "No active holds",
                    body = "When you accept a package hold request, it will appear here.",
                    ctaLabel = "Browse open requests",
                    onCta = { navController.navigate(Routes.CREATE) },
                )
            }
        } else {
            items(activeHolds, key = { it.id }) { shipment ->
                ShipmentCard(
                    shipment = shipment,
                    onClick = { navController.navigate(Routes.shipmentDetail(shipment.id)) },
                )
            }
        }
        item {
            Text(
                text = "Quick Actions",
                color = c.textPrimary,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
            )
        }
        item {
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = c.surface),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
            ) {
                Column {
                    ActionRow(icon = Icons.Outlined.LocationOn, tint = c.accent, label = "Find nearby requests") {
                        navController.navigate(Routes.CREATE)
                    }
                    Spacer(modifier = Modifier.height(0.dp))
                    ActionRow(icon = Icons.Outlined.Paid, tint = c.success, label = "My earnings") {
                        navController.navigate(Routes.CREATE)
                    }
                    Spacer(modifier = Modifier.height(0.dp))
                    ActionRow(icon = Icons.Outlined.VerifiedUser, tint = c.warmOrange, label = "Get verified") {
                        navController.navigate(Routes.CREATE)
                    }
                }
            }
        }
    }
}

@Composable
private fun EarningsCard() {
    val c = PorchivoTheme.colors
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = c.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(14.dp),
        ) {
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .background(c.successSoft, CircleShape),
                contentAlignment = Alignment.Center,
            ) {
                Icon(
                    imageVector = Icons.Outlined.Paid,
                    contentDescription = null,
                    tint = c.success,
                    modifier = Modifier.size(22.dp),
                )
            }
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "Potential: $80–$250/mo",
                    color = c.textPrimary,
                    fontSize = 17.sp,
                    fontWeight = FontWeight.Black,
                )
                Text(
                    text = "Hold packages for neighbors · Keep 85% · 2-day payout",
                    color = c.textSecondary,
                    fontSize = 12.sp,
                )
            }
        }
    }
}

@Composable
private fun ActionRow(
    icon: ImageVector,
    tint: androidx.compose.ui.graphics.Color,
    label: String,
    onClick: () -> Unit,
) {
    val c = PorchivoTheme.colors
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 14.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = tint,
            modifier = Modifier.size(18.dp),
        )
        Text(
            text = label,
            color = c.textPrimary,
            fontSize = 15.sp,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier.weight(1f),
        )
        Icon(
            imageVector = Icons.Filled.ChevronRight,
            contentDescription = null,
            tint = c.textMuted,
            modifier = Modifier.size(18.dp),
        )
    }
}

File: ios/Porchivo/Views/PorchPartnerScreen.swift
//
//  PorchPartnerScreen.swift
//  Porchivo
//
//  Free tier tab — active holds, nearby requests, quick actions for
//  Porch Partner users. Shown when user has no active org membership.
//

import SwiftUI

struct PorchPartnerScreen: View {
    @Environment(AppState.self) private var appState
    @Environment(\.porchivo) private var c
    @State private var path = NavigationPath()

    var body: some View {
        NavigationStack(path: $path) {
            ScrollView {
                VStack(spacing: 16) {
                    header
                    earningsCard
                    activeHoldsSection
                    quickActions
                }
                .padding(.horizontal, 16)
                .padding(.top, 16)
                .padding(.bottom, 24)
            }
            .background(c.background.ignoresSafeArea())
            .navigationTitle("Porch Partner")
            .navigationDestination(for: Route.self) { route in
                RouteView(route: route, path: $path)
            }
        }
    }

    private var header: some View {
        VStack(spacing: 4) {
            Text("Earn by helping neighbors")
                .font(.system(size: 16, weight: .semibold))
                .foregroundStyle(c.textSecondary)
                .frame(maxWidth: .infinity, alignment: .leading)
        }
    }

    private var earningsCard: some View {
        HStack(spacing: 14) {
            ZStack {
                Circle().fill(c.successSoft)
                Image(systemName: "dollarsign.circle.fill")
                    .font(.system(size: 22, weight: .bold))
                    .foregroundStyle(c.success)
            }
            .frame(width: 48, height: 48)

            VStack(alignment: .leading, spacing: 2) {
                Text("Potential: $80–$250/mo")
                    .font(.system(size: 17, weight: .black))
                    .foregroundStyle(c.textPrimary)
                Text("Hold packages for neighbors · Keep 85% · 2-day payout")
                    .font(.system(size: 12))
                    .foregroundStyle(c.textSecondary)
            }
            Spacer()
        }
        .padding(16)
        .background(c.surface, in: .rect(cornerRadius: Radius.lg))
        .shadow(color: c.textPrimary.opacity(0.05), radius: 8, y: 3)
    }

    private var activeHoldsSection: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("Active Holds")
                .font(.system(size: 18, weight: .bold))
                .foregroundStyle(c.textPrimary)

            if appState.shipments.filter({ $0.status == .accepted }).isEmpty {
                EmptyState(
                    symbol: "shippingbox.fill",
                    title: "No active holds",
                    message: "When you accept a package hold request, it will appear here.",
                    ctaLabel: "Browse open requests"
                ) {
                    path.append(Route.create)
                }
            } else {
                ForEach(appState.shipments.filter { $0.status == .accepted }) { shipment in
                    NavigationLink(value: Route.shipmentDetail(shipment.id)) {
                        ShipmentCard(shipment: shipment)
                    }
                    .buttonStyle(.plain)
                }
            }
        }
    }

    private var quickActions: some View {
        VStack(spacing: 0) {
            actionRow("Find nearby requests", "location.fill", c.accent) {
                path.append(Route.create)
            }
            Divider().overlay(c.border).padding(.leading, 54)
            actionRow("My earnings", "banknote.fill", c.success) {
                path.append(Route.create)
            }
            Divider().overlay(c.border).padding(.leading, 54)
            actionRow("Get verified", "checkmark.shield.fill", c.warmOrange) {
                path.append(Route.create)
            }
        }
        .background(c.surface, in: .rect(cornerRadius: Radius.lg))
        .shadow(color: c.textPrimary.opacity(0.05), radius: 6, y: 2)
    }

    private func actionRow(_ label: String, _ symbol: String, _ tint: Color, action: @escaping () -> Void) -> some View {
        Button(action: { Haptics.selection(); action() }) {
            HStack(spacing: 12) {
                Image(systemName: symbol)
                    .font(.system(size: 14, weight: .bold))
                    .foregroundStyle(tint)
                    .frame(width: 28)
                Text(label)
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundStyle(c.textPrimary)
                Spacer()
                Image(systemName: "chevron.right")
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(c.textMuted)
            }
            .padding(.horizontal, 14)
            .padding(.vertical, 14)
        }
        .buttonStyle(.plain)
    }
}

#Preview {
    PorchPartnerScreen().environment(AppState())
}
